gnomad_qc/v4/sample_qc/generate_qc_mt.py

```diff
@@ -0,0 +1,293 @@
+    mt = annotate_adj(mt)
+
+    cp_path = get_checkpoint_path("combined_pre_ld_prune_qc_mt", mt=True)
+    # TODO: As I have it now, v3 and v4 are used to determine AF and callrate, is that OK, or should they be considered
+    #  independently and then shared passing variants merged?
+    mt = get_qc_mt(
+        mt,
+        bi_allelic_only=bi_allelic_only,
+        min_af=min_af,
+        min_callrate=min_callrate,
+        min_inbreeding_coeff_threshold=min_inbreeding_coeff_threshold,
+        min_hardy_weinberg_threshold=None,  # Already filtered from the initial set of QC variants
+        apply_hard_filters=False,
+        ld_r2=ld_r2,
+        checkpoint_path=cp_path,
+        filter_lcr=False,  # Already filtered from the initial set of QC variants
+        filter_decoy=False,  # Doesn't exist for hg38
+        filter_segdup=False,  # Already filtered from the initial set of QC variants
+    )
+    logger.info(
+        "Number of pre LD-pruned QC sites in gnomAD v3 + v4: %d...",
+        hl.read_matrix_table(cp_path).count_rows(),
+    )
+
+    return mt
+
+
+def main(args):
+    hl.init(
+        log="/generate_qc_mt.log",
+        default_reference="GRCh38",
+        tmp_dir="gs://gnomad-tmp-4day",
+    )
+    # NOTE: remove this flag when the new shuffle method is the default
+    hl._set_flags(use_new_shuffle="1")
+
+    overwrite = args.overwrite
+    ld_r2 = args.ld_r2
+    test = args.test
+    n_partitions = args.n_partitions
+
+    try:
+        if args.create_v3_filtered_dense_mt:
+            # Note: This command removes hard filtered samples
+            mt = get_gnomad_v3_mt(key_by_locus_and_alleles=True)
+            mt = create_filtered_dense_mt(mt, test, split=True)
+            mt = mt.checkpoint(
+                get_dense_predetermined_qc(version="3.1", test=test).path,
+                overwrite=overwrite,
+            )
+            logger.info(
+                "Number of predetermined QC variants found in the gnomAD v3 MatrixTable: %d...",
+                mt.count_rows(),
+            )
+
+        if args.create_v4_filtered_dense_mt:
+            # Note: This subset dense MatrixTable was created before the final hard filtering was determined
+            # Hard filtering is performed in `generate_qc_mt` before applying variant filters
+            vds = get_gnomad_v4_vds(split=True, remove_hard_filtered_samples=False)
+            mt = create_filtered_dense_mt(vds, test)
+            mt = mt.checkpoint(
+                get_dense_predetermined_qc(test=test).path, overwrite=overwrite
+            )
+            logger.info(
+                "Number of predetermined QC variants found in the gnomAD v4 VDS: %d...",
+                mt.count_rows(),
+            )
+
+        if args.generate_qc_mt:
+            v3_mt = hl.read_matrix_table(
+                get_dense_predetermined_qc(version="3.1", test=test).path,
+                _n_partitions=n_partitions,
```

Comment:
I tested this, and you are right, it will write out double the partitions. I can either make a note warning about this, or I could also add a naive_coalesce before writing.

Reply:
Another options is  split the passed arg early on and use the result when pulling in the predetermined qc MT, e.g n_partitions=args.n_partitions/2. You'd end up with the desired partitions. Naive_coalesce also makes sense to me but I'm not sure of its impact on runtime and if we are concerned with that.

Reply:
Additionally, is half the desired partitions too small for the predetermined QC MT?

Reply:
The MT isn't that big since the number of variants will be less that 250K, so I think either args.n_partitions/2 or naive_coalesce will work. However, Kristen just added in a repartition step to before the ld-prune when making the QC MT (get_qc_mt) So I'm thinking the solution is to just pass this into that function

Reply:
That makes sense to me too